lib/modules/datasource/azure-bicep-resource/index.ts

```diff
@@ -0,0 +1,84 @@
+import { cache } from '../../../util/cache/package/decorator';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, ReleaseResult } from '../types';
+import { BicepTypeIndex } from './schema';
+
+const BICEP_TYPES_INDEX_URL =
+  'https://raw.githubusercontent.com/Azure/bicep-types-az/main/generated/index.json';
+
+export class AzureBicepResourceDatasource extends Datasource {
+  static readonly id = 'azure-bicep-resource';
+
+  override readonly defaultConfig = {
+    commitMessageTopic: 'resource {{depName}}',
+    commitMessageExtra: 'to {{{newVersion}}}',
+    branchTopic: '{{{depNameSanitized}}}-{{{newVersion}}}',
+    prBodyColumns: ['Resource', 'Change'],
+    prBodyDefinitions: {
+      Resource: '{{{depNameLinked}}}',
+    },
+  };
+
+  constructor() {
+    super(AzureBicepResourceDatasource.id);
+  }
+
+  @cache({
+    namespace: `datasource-${AzureBicepResourceDatasource.id}`,
+    key: ({ packageName }: GetReleasesConfig) => `getReleases-${packageName}`,
+  })
+  async getReleases(
+    getReleasesConfig: GetReleasesConfig
+  ): Promise<ReleaseResult | null> {
+    const { packageName } = getReleasesConfig;
+
+    const resourceVersionIndex = await this.getResourceVersionIndex();
+    const versions = resourceVersionIndex[packageName.toLowerCase()];
+
+    if (!versions) {
+      return null;
+    }
+
+    const firstSlashIndex = packageName.indexOf('/');
+    const namespaceProvider = packageName
+      .slice(0, firstSlashIndex)
+      .toLowerCase();
+    const type = packageName.slice(firstSlashIndex + 1).toLowerCase();
+
+    return {
+      releases: versions.map((x) => ({
+        version: x,
+        changelogUrl: `https://learn.microsoft.com/en-us/azure/templates/${namespaceProvider}/change-log/${type}#${x}`,
+      })),
```

Comment:
```suggestion
      releases: versions.map((version) => ({
        version:,
        changelogUrl: `https://learn.microsoft.com/en-us/azure/templates/${namespaceProvider}/change-log/${type}#${version}`,
      })),
```
use descriptive variables